frontend/rust-lib/flowy-database2/src/services/database_view/view_editor.rs

```diff
@@ -307,33 +307,46 @@ impl DatabaseViewEditor {
         let rows = vec![Arc::new(row.clone())];
         let mut rows = self.v_filter_rows(rows).await;
 
-        if let Some(row) = rows.pop() {
-          let result = controller.did_update_group_row(old_row, &row, &field);
-
-          if let Ok(result) = result {
-            let mut group_changes = GroupChangesPB {
-              view_id: self.view_id.clone(),
-              ..Default::default()
-            };
-            if let Some(inserted_group) = result.inserted_group {
-              tracing::trace!("Create group after editing the row: {:?}", inserted_group);
-              group_changes.inserted_groups.push(inserted_group);
-            }
-            if let Some(delete_group) = result.deleted_group {
-              tracing::trace!("Delete group after editing the row: {:?}", delete_group);
-              group_changes.deleted_groups.push(delete_group.group_id);
-            }
+        let mut group_changes = GroupChangesPB {
+          view_id: self.view_id.clone(),
+          ..Default::default()
+        };
 
-            if !group_changes.is_empty() {
-              notify_did_update_num_of_groups(&self.view_id, group_changes).await;
-            }
+        let (inserted_group, deleted_group, row_changesets) = if let Some(row) = rows.pop() {
+          if let Ok(result) = controller.did_update_group_row(old_row, &row, &field) {
+            (
+              result.inserted_group,
+              result.deleted_group,
+              result.row_changesets,
+            )
+          } else {
+            (None, None, vec![])
+          }
+        } else {
+          if let Ok(result) = controller.did_delete_row(&row) {
```

Comment:
@appflowy must the group controller have all of the rows, including hidden ones? If so we need to store that info there. Here I simply remove it.